### PR TITLE
[WIP] Skylls - Skyrat skills and stats system

### PR DESCRIPTION
**Pull request by BobJoga**
## About The Pull Request

TLDR skill and stat system based on GURPS, inspired by the forbidden server and ripped off from IS12 warfare code but i'll try to tone down implementation to keep it fun.

Very work in progress. Screenshots and stuff coming soon.

### Implementation
If something is "Implemented", it means it has been integrated with at least one system in the game. Otherwise, nothing uses the stat or skill at all.

**Current skills planned, ticked ones have been at least partially implemented:**
- [ ] Melee combat
- [ ] Ranged combat
- [ ] First aid
- [ ] Surgery
- [ ] Construction
- [ ] Electronics
- [ ] Gaming

**Current stats planned, ticked ones have been at least partially implemented:**
- [ ] Endurance
- [ ] Intelligence
- [ ] Strength
- [ ] Dexterity

### How will it work?
Mood will affect skill and stat based actions greatly, thus will no longer be a useless background mechanic.
Skills and stats are set on the character setup. Stats are the same distribution of points for everyone, but skills can be changed for each role, each role has a set of skill points being given out based on how important of a role it is (captain gets a lot of skill points, assistants get little to none).
As such, skills and stats are mostly set by the player, the job only really affects how much you get and how easy it is to level up ingame (which is rare and not incentivized).

## Why It's Good For The Game

Mechanical enforcement of skills, prevents know-all characters and forces people to choose what they specialize themselves in, thus making the game more apt for roleplay.

## Changelog
:cl:
add: Implemented a proper skill and stat system. Most of it should be pretty intuitive on how it works.
/:cl:

### CODE REVIEW

**Comment by NotRanged** [comment]
>Mood will affect skill and stat based actions greatly, thus will no longer be a useless background mechanic.

-1

**Comment by BobJoga** [comment]
> > Mood will affect skill and stat based actions greatly, thus will no longer be a useless background mechanic.
> 
> -1

too bad

**Comment by Jeff-Dome** [comment]
Ngl, I don't believe this is inspired by, given it's literally a rip from the code, but however. don't care plus I coomed.

**Comment by BobJoga** [comment]
> Ngl, I don't believe this is inspired by, given it's literally a rip from the code, but however. don't care plus I coomed.

tbh i'm gonna edit a bit since yea its pretty much stolen

**Comment by MLGTASTICa** [comment]
Great change +1 +1 , this will promote RP and prevent assistants from rushing tators for their loot or for people to do others jobs , this also solves the issue of metaknowdlege.

**Comment by ReynTheLord** [comment]
Personally, I believe, one, MLGTASTIC is being sarcastic. However

-1.

This will, not might, WILL, lead to stastistic minmaxing, and to some people taking certain roles just for stats more often. This also leads to skilled players just being fucked when they go antag, because sec WILL have higher stats, and mood systems are... Fucky at best.

Forgive my language.

Some people will simply just use this to powergame, increasing the problems relating to powergame, And has NOTHING to do with metaknowledge. 

I may be wrong, but, In my honest opinion, this is more trouble than its worth.

**Comment by BobJoga** [comment]
> Personally, I believe, one, MLGTASTIC is being sarcastic. However
> 
> -1.
> 
> This will, not might, WILL, lead to stastistic minmaxing, and to some people taking certain roles just for stats more often. This also leads to skilled players just being fucked when they go antag, because sec WILL have higher stats, and mood systems are... Fucky at best.
> 
> Forgive my language.
> 
> Some people will simply just use this to powergame, increasing the problems relating to powergame, And has NOTHING to do with metaknowledge.
> 
> I may be wrong, but, In my honest opinion, this is more trouble than its worth.
> 
> Know all characters should be prevented by roleplay policy, not by making them statisticly weaker.

Roleplay policy is a joke. Preventing abuse mechanically is the only true solution.

**Comment by Jeff-Dome** [comment]
>rules
M'kay. you... you really think a man yelling at someone who can boot up a VM software with a different CID and a Different IP (Lol owned) can stop someone for using le metaknowledge....
>the reddit/discordmain main on reality unfolding.

**Comment by NotRanged** [comment]
> > rules
> > M'kay. you... you really think a man yelling at someone who can boot up a VM software with a different CID and a Different IP (Lol owned) can stop someone for using le metaknowledge....
> > the reddit/discordmain main on reality unfolding.

this reads like it was generated by a markov chain bot

**Comment by ThatGuyFromThatPlace** [comment]
> Personally, I believe, one, MLGTASTIC is being sarcastic. However
> 
> -1.
> 
> This will, not might, WILL, lead to stastistic minmaxing, and to some people taking certain roles just for stats more often. This also leads to skilled players just being fucked when they go antag, because sec WILL have higher stats, and mood systems are... Fucky at best.
> 
> Forgive my language.
> 
> Some people will simply just use this to powergame, increasing the problems relating to powergame, And has NOTHING to do with metaknowledge.
> 
> I may be wrong, but, In my honest opinion, this is more trouble than its worth.
> 
> Know all characters should be prevented by roleplay policy, not by making them statisticly weaker.

Playing a game shouldn't be an exercise in dancing around rules, and, I'm a firm believer that, you shouldn't punish people for stuff that's mechanically possible for someone playing in good faith to do (Which, is why I think the whole Combat mode/Combat indicator requirement is kinda dumb, because, it could just be coded that you need combat mode on to attack people instead of it being a simple rule, but, hey, that'd only make sense)
Relying on some OOC rulebook and making a minefield for players who have no poor intentions is poor game design and using an aggressive and poor ruleset to make up for it.

**Comment by MLGTASTICa** [comment]
> Personally, I believe, one, MLGTASTIC is being sarcastic. However
> 
> -1.
> 
> This will, not might, WILL, lead to stastistic minmaxing, and to some people taking certain roles just for stats more often. This also leads to skilled players just being fucked when they go antag, because sec WILL have higher stats, and mood systems are... Fucky at best.
> 
> Forgive my language.
> 
> Some people will simply just use this to powergame, increasing the problems relating to powergame, And has NOTHING to do with metaknowledge.
> 
> I may be wrong, but, In my honest opinion, this is more trouble than its worth.
> 
> Know all characters should be prevented by roleplay policy, not by making them statisticly weaker.

Im not being sarcastic , this is a much needed change,this is a literal cap to how much people can powergame ontop of providing a good and reliable fix to metaknowledge (people that know every job for some reason) . Roleplay policy is not good and not enforcing it mechanically is a big mistake.

**Comment by Owai-Seek** [comment]
I don't mind having to set skills being set so powergame can be reduced, but tying everything to a terrible imbalanced system like mood is ridiculous IMO.

**Comment by MLGTASTICa** [comment]
> I don't mind having to set skills being set so powergame can be reduced, but tying everything to a terrible imbalanced system like mood is ridiculous IMO.

Mind argumenting your point ? i don't see how mood is broken.

**Comment by KathrinBailey** [comment]
> > I don't mind having to set skills being set so powergame can be reduced, but tying everything to a terrible imbalanced system like mood is ridiculous IMO.
> 
> Mind argumenting your point ? i don't see how mood is broken.

Mood has too few moods.
Just eat and get someone to click you on help intent every so often.

**Comment by MLGTASTICa** [comment]
> > > I don't mind having to set skills being set so powergame can be reduced, but tying everything to a terrible imbalanced system like mood is ridiculous IMO.
> > 
> > 
> > Mind argumenting your point ? i don't see how mood is broken.
> 
> Mood has too few moods.
> Just eat and get someone to click you on help intent every so often.

Yes but that only gets you so far . Achieving the maximum mood is preety hard , getting someone to click on you every so few minutes is tedious and rarely done , and the eat mood is only achieved from cook food.

**Comment by Jeff-Dome** [comment]
> > > > I don't mind having to set skills being set so powergame can be reduced, but tying everything to a terrible imbalanced system like mood is ridiculous IMO.
> > > 
> > > 
> > > Mind argumenting your point ? i don't see how mood is broken.
> > 
> > 
> > Mood has too few moods.
> > Just eat and get someone to click you on help intent every so often.
> 
> Yes but that only gets you so far . Achieving the maximum mood is preety hard , getting someone to click on you every so few minutes is tedious and rarely done , and the eat mood is only achieved from cook food.

It legit, the easiest thing to do to get the highest mood. wym it's pretty fucking hard.

delusional

**Comment by MLGTASTICa** [comment]
> > > > > I don't mind having to set skills being set so powergame can be reduced, but tying everything to a terrible imbalanced system like mood is ridiculous IMO.
> > > > 
> > > > 
> > > > Mind argumenting your point ? i don't see how mood is broken.
> > > 
> > > 
> > > Mood has too few moods.
> > > Just eat and get someone to click you on help intent every so often.
> > 
> > 
> > Yes but that only gets you so far . Achieving the maximum mood is preety hard , getting someone to click on you every so few minutes is tedious and rarely done , and the eat mood is only achieved from cook food.
> 
> It legit, the easiest thing to do to get the highest mood. wym it's pretty fucking hard.
> 
> delusional

Highest mood is "My mind feels like a temple" , codewise this is +12 , eating + headpat only gets you to +6 , which is "Amazing"

**Comment by Jeff-Dome** [comment]
> > > > > > I don't mind having to set skills being set so powergame can be reduced, but tying everything to a terrible imbalanced system like mood is ridiculous IMO.
> > > > > 
> > > > > 
> > > > > Mind argumenting your point ? i don't see how mood is broken.
> > > > 
> > > > 
> > > > Mood has too few moods.
> > > > Just eat and get someone to click you on help intent every so often.
> > > 
> > > 
> > > Yes but that only gets you so far . Achieving the maximum mood is preety hard , getting someone to click on you every so few minutes is tedious and rarely done , and the eat mood is only achieved from cook food.
> > 
> > 
> > It legit, the easiest thing to do to get the highest mood. wym it's pretty fucking hard.
> > delusional
> 
> Highest mood is "My mind feels like a temple" , codewise this is +12 , eating + headpat only gets you to +6 , which is "Amazing"

Or you could just smoke, be full along with eating cooked food... it's really not that hard.

**Comment by BobJoga** [comment]
this won't work here
